url8752fbdeb0_sixtam_args_tgz-pJ8-cz_mff_dpp_args_examples_ConstraintTestJ8/src/cz/mff/dpp/args/ConstraintUtils.java

```diff
@@ -0,0 +1,494 @@
+
+		if (constraint.min().isEmpty()) {
+			return true;
+		}
+
+		Object min;
+		try {
+			min = ReflectUtils.valueFromString(type, constraint.min());
+		} catch (ConfException ex) {
+			throw new ConfException(ex,
+					"Cannot create value for min from '%s'", constraint.min());
+		}
+
+		int result = compare(constraint, type, value, min);
+
+		if (result < 0) {
+			return false;
+		} else {
+			return true;
+		}
+
+	}
+
+	/**
+	 * 
+	 * Checks whether an object is less or equal to required maximum.
+	 * 
+	 * @param constraint
+	 *            the corresponding {@code Constraint}
+	 * @param type
+	 *            type of the object
+	 * @param value
+	 *            value to check
+	 * 
+	 * @return {@code true} if an object passed the test or constraint is not
+	 *         specified, otherwise {@code false}
+	 * 
+	 * @throws CheckException
+	 *             if object does not pass test
+	 * @throws ConfException
+	 *             if needed objects values cannot be created, appropriate
+	 *             functions called
+	 * 
+	 * @see #checkValue(AccessibleObject, Object)
+	 * @see Constraint
+	 */
+	private static boolean checkMax(Constraint constraint, Class<?> type,
+			Object value) throws CheckException, ConfException {
+
+		if (constraint.max().isEmpty()) {
+			return true;
+		}
+
+		Object max;
+		try {
+			max = ReflectUtils.valueFromString(type, constraint.max());
+		} catch (ConfException ex) {
+			throw new ConfException(ex, "Cannot create value for max from %s",
+					constraint.min());
+		}
+
+		int result = compare(constraint, type, value, max);
+
+		if (result > 0) {
+			return false;
+		} else {
+			return true;
+		}
+
+	}
+
+	/**
+	 * 
+	 * Compares two objects.
+	 * <p>
+	 * Values are compared in the following way (in order of execution):
+	 * <ul>
+	 * <li>{@code static compare(T x, T y)} of the object (or of comparator if
+	 * specified)</li>
+	 * <li>{@code compare.compareTo(compareTo)}</li>
+	 * </ul>
+	 * </p>
+	 * 
+	 * @param constraint
+	 *            associated constraint
+	 * @param type
+	 * @param compare
+	 *            value
+	 * @param compareTo
+	 *            value to compare to
+	 * @return result of comparison
+	 * @throws ConfException
+	 *             if objects cannot be compared.
+	 */
+	private static int compare(Constraint constraint, Class<?> type,
+			Object compare, Object compareTo) throws ConfException {
+
+		Class<?>[] compareToArgsTypes = { type };
+
+		// let's try static comapre(x, y) on creatorType
+		Class<?>[] compareArgsTypes = { type, type };
+
+		Class<?> comparatorType;
+
+		if (constraint.comparator() != None.class) {
+			comparatorType = constraint.comparator();
+		} else {
+			comparatorType = FactoryRegistry.getFactoryType(type);
+		}
+
+		try {
+			Method comparator = ReflectUtils.getMethod(comparatorType, COMPARE,
+					compareArgsTypes);
+
+			if (ReflectUtils.isFieldAssignableFrom(int.class, comparator)) {
+				return (Integer) comparator.invoke(null, compare, compareTo);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)